digits/model/images/classification/views.py

```diff
@@ -190,11 +190,8 @@ def create():
                         elif epoch == -1:
                             pretrained_model = old_job.train_task().pretrained_model
                         else:
-                            for filename, e in old_job.train_task().snapshots:
-                                if e == epoch:
-                                    pretrained_model = filename
-                                    break
-
+                            # verify snapshot exists
+                            pretrained_model = old_job.train_task().get_snapshot(epoch, download=True)
```

Comment:
Why this change? It's better to throw a `BadRequest` than the `ValueError` that gets thrown if the epoch is missing unless there's a reason not to.